### DISCUSSION TITLE
Plastic flow at corners

### DISCUSSION THREAD

**Discussion by superflyer11** [thelfer/tfel#642] [Q&A] [answered]
Are there any examples of dealing with corners/singularities for plastic flow?

**Comment by thelfer** [comment]
It depends on the singularity you want to treat. What have you in mind ?

**Reply by superflyer11** [reply]
A simple one like https://thelfer.github.io/tfel/web/drucker-prager-cap.html at the cone tip as a start

**Reply by thelfer** [reply, marked as answer]
The apex is always a problem. There are two classical solutions:

- one given by Neto Souza in https://onlinelibrary.wiley.com/doi/book/10.1002/9780470694626
- one being a regularisation: see https://thelfer.github.io/tfel/web/MohrCoulomb.html

**Reply by superflyer11** [reply]
I am trying to fit a paraboloid to the cone like this:
https://www.desmos.com/3d/ortc6732tg

And I have two mfront behaviours, one for the cone and one for the paraboloid:
[DP.mfront.txt](https://github.com/user-attachments/files/17654303/DP.mfront.txt)
[DP_paraboloidal.mfront.txt](https://github.com/user-attachments/files/17654302/DP_paraboloidal.mfront.txt)

Uniaxial compression test from the origin:
[DP_TriaxialTest_copy.py.txt](https://github.com/user-attachments/files/17654379/DP_TriaxialTest_copy.py.txt)

And this is the output, I expected the yield surface to be slightly larger in the paraboloidal surface but instead it is halved by two
![image](https://github.com/user-attachments/assets/b638ca2a-c389-4c15-9154-2e044b69837f)
![image](https://github.com/user-attachments/assets/be2dcfaf-0a36-49d6-83ff-05cf73ffe705)

It must be some math error perhaps in the behaviour file that I am missing but I cannot spot it

And as a spot check would this derivative be correct?

![image](https://github.com/user-attachments/assets/d8b4642e-0448-409b-b159-553beb7acf06)
![image](https://github.com/user-attachments/assets/bff12177-aaaa-49ce-a23d-af4966bdda75)
![image](https://github.com/user-attachments/assets/4a5aafa5-3d85-4f3f-bd99-3c92993fbb7c)

**Reply by superflyer11** [reply]
I ended up using a hyperboloidal form
![image](https://github.com/user-attachments/assets/e8717df3-0e05-4e26-9e8f-749b2b520caf)
as a approaches zero it becomes a closer fit, and at a=0 it becomes a cone

To simulate a situation with c=0, set c close to 0 but set a to be very small such that the initial stress state is still within the hyperboloid

**Reply by thelfer** [reply]
@superflyer11 Sorry I was attempting a conference (https://mealor2.sciencesconf.org/resource/page/id/7)
Is everything working now ?

**Reply by superflyer11** [reply]
@thelfer Interesting! 
This should not be an issue for now. 

Onto a side tangent, when I try to implement associative isotropic hardening on Drucker-Prager I find it difficult.

in von Mises the rate of equivalent plastic strain is equal to the rate of the plastic multiplier
![image](https://github.com/user-attachments/assets/60727850-deb1-4560-a9a3-4cb87cef4864)
but I cannot find any derivation for more advanced models
therefore I simply tried adding in the term `H*p` into the yield function of Drucker-Prager
[DP_hyperboloidal_hardening.mfront.txt](https://github.com/user-attachments/files/17692494/DP_hyperboloidal_hardening.mfront.txt)
[2_DP_test.zip](https://github.com/user-attachments/files/17692528/2_DP_test.zip)

which produces a result that I don't understand
![image](https://github.com/user-attachments/assets/00245055-a305-42ae-94cf-50a2e5e6164b)
